src/main/java/com/example/demo/services/UsersService.java

```diff
@@ -125,12 +141,382 @@ public Integer extractData(ResultSet rs) throws SQLException, DataAccessExceptio
+		singleUserAdvanceDetailsBO.setAddress(userBasicDetailsBO.getAddress());
+		singleUserAdvanceDetailsBO.setBirthDate(userBasicDetailsBO.getBirthDate());
+		singleUserAdvanceDetailsBO.setMaritalStatus(userBasicDetailsBO.getMaritalStatus());
+		singleUserAdvanceDetailsBO.setAdhar(userBasicDetailsBO.getAdhar());
+		singleUserAdvanceDetailsBO.setReligion(userBasicDetailsBO.getReligion());
+		singleUserAdvanceDetailsBO.setCaste(userBasicDetailsBO.getCaste());
+		singleUserAdvanceDetailsBO.setNationality(userBasicDetailsBO.getNationality());
+		singleUserAdvanceDetailsBO.setGender(userBasicDetailsBO.getGender());
+		singleUserAdvanceDetailsBO.setAlternateMobile(userBasicDetailsBO.getAlternateMobile());
+	}
+	public String getClassNameFromClassID(String classID)
+	{
+		String query = "SELECT class_name FROM classes where class_id=?";	
+		return jdbcTemplate.query(query, new PreparedStatementSetter() {
+			
+			@Override
+			public void setValues(PreparedStatement ps)throws SQLException
+			{
+				ps.setString(1, classID);
+			}
+		}, new ResultSetExtractor<String>() {
+			
+			@Override
+			public String extractData(ResultSet rs) throws SQLException, DataAccessException {
+				
+				String classes=null;
+				while (rs.next()) 
+				{
+					classes = rs.getString("class_name");
+					
+				}
+				return classes;
+			}
+			
+		});
+	}
+	public String getSubjectNameFromSubjectID(String subjectID)
+	{
+		String query = "SELECT sub_name FROM subjects where sub_id=?";	
+		return jdbcTemplate.query(query, new PreparedStatementSetter() {
+			
+			@Override
+			public void setValues(PreparedStatement ps)throws SQLException
+			{
+				ps.setString(1, subjectID);
+			}
+		}, new ResultSetExtractor<String>() {
+			
+			@Override
+			public String extractData(ResultSet rs) throws SQLException, DataAccessException {
+				
+				String subjects=null;
+				while (rs.next()) 
+				{
+					subjects = rs.getString("sub_name");
+					
+				}
+				return subjects;
+			}
+			
+		});
+	}
+	public String getMonthNameFromMonthMaster(String monthId)
+	{
+		String query = "SELECT month FROM month_master where month_id=?";	
+		return jdbcTemplate.query(query, new PreparedStatementSetter() {
+			
+			@Override
+			public void setValues(PreparedStatement ps)throws SQLException
+			{
+				ps.setString(1, monthId);
+			}
+		}, new ResultSetExtractor<String>() {
+			
+			@Override
+			public String extractData(ResultSet rs) throws SQLException, DataAccessException {
+				
+				String month=null;
+				while (rs.next()) 
+				{
+					month = rs.getString("month");
+					
+				}
+				return month;
+			}
+			
+		});
+	}
+	
+	//Prototype to get the user academic details in UserAdvanceDeails
+	//private Map<String, UserAcademicDetailsBO> userAcademicDetails; // Map of Academic id to Academic Details
+	public Map<String, UserAcademicDetailsBO> getAcademicDetailsOfOneUser(String userId)
+	{
+		String query = "SELECT * FROM USER_ACADEMIC_DETAILS where user_id=?";
```

Comment:
write a query in a such a way that it will fetch class name and subject name in same query. And remove those methods added for fetching class name and subject name. **Please refer API developed by me for fetching Studen details, you will get an Idea**

Reply:
need to revisit